### PR TITLE
Better enable split mesh for multiapp configurations

### PR DESCRIPTION
**Pull request by loganharbour**
refs #17318

### CODE REVIEW

**Comment by loganharbour** [comment]
I swore my default was a draft... sorry folks.

**Comment by loganharbour** [comment]
@lindsayad if you wouldn't mind

File: test/tests/mesh/splitting/tests [hunk 1]
```diff
@@ -12,18 +12,33 @@
...
+      max_parallel = 3
+
+      detail = 'set by command line'
+    []
+    [from_input]
+      type = 'Exodiff'
+      input = 'simple_diffusion.i'
+      exodiff = 'simple_diffusion_out.e'
+      cli_args = 'Mesh/use_split=true Mesh/split_file=foo UserObjects/splittester/type=SplitTester'
+      prereq = 'make_split'
```
Comment (lindsayad) [inline test/tests/mesh/splitting/tests:36]:
```suggestion
      prereq = 'use_split/from_cli
```

File: test/tests/mesh/splitting/tests [hunk 2]
```diff
@@ -145,7 +160,7 @@
 
     min_parallel = 3
     max_parallel = 3
-    prereq = 'use_split'
+    prereq = 'use_split/from_cli'
```
Comment (lindsayad) [inline test/tests/mesh/splitting/tests:163]:
```suggestion
    prereq = 'use_split/from_input'
```

**Review by lindsayad — CHANGES_REQUESTED** [review]

**Review by lindsayad — APPROVED** [review]

**Review by GiudGiud — APPROVED** [review]
awesome. I think this finishes the work for the milestone but we can discuss that more

**Pull request merged**